sdk/tests/conformance2/transform_feedback/simultaneous_binding.html

```diff
@@ -0,0 +1,216 @@
+
+const uniformBuffer = createBuffer(gl, new Float32Array([1, 0, 0, 0]));
+const ubi = gl.getUniformBlockIndex(prog, "UniformBlock");
+gl.uniformBlockBinding(prog, ubi, 0);
+gl.bindBufferBase(gl.UNIFORM_BUFFER, 0, uniformBuffer);
+
+drawWithFeedbackBound(gl, prog, vao, tf, true);
+wtu.glErrorShouldBe(gl, gl.NO_ERROR, "transform feedback should be successful");
+
+const expected = [2, 4, 6, 8];
+gl.bindBuffer(gl.TRANSFORM_FEEDBACK_BUFFER, tfBuffer);
+wtu.checkFloatBuffer(gl, gl.TRANSFORM_FEEDBACK_BUFFER, expected);
+
+debug("<hr/>Test ARRAY_BUFFER");
+// this should fail because the transform feedback's buffer #0 and the
+// badVao's buffer #0 are the same buffer
+const badVao = createVAO(gl, tfBuffer);
+drawWithFeedbackBound(gl, prog, badVao, tf, false);
+wtu.glErrorShouldBe(gl, gl.INVALID_OPERATION, "buffer used as vertex attrib and tf simultaneously");
+drawWithFeedbackBound(gl, prog, badVao, tf, true);
+wtu.glErrorShouldBe(gl, gl.INVALID_OPERATION, "buffer used as vertex attrib and tf simultaneously");
+gl.bindVertexArray(vao);
+
+gl.bindBuffer(gl.TRANSFORM_FEEDBACK_BUFFER, tfBuffer);
+wtu.checkFloatBuffer(gl, gl.TRANSFORM_FEEDBACK_BUFFER, expected, "should be the same as before as nothing has executed");
+
+debug("<hr/>Test UNIFORM_BUFFER");
+gl.bindBufferBase(gl.UNIFORM_BUFFER, 0, tfBuffer);
+gl.bindBuffer(gl.UNIFORM_BUFFER, null); // tfBuffer is still bound at index 0
+drawWithFeedbackBound(gl, prog, vao, tf, false);
+wtu.glErrorShouldBe(gl, gl.INVALID_OPERATION, "buffer used as uniform buffer and tf simultaneously");
+drawWithFeedbackBound(gl, prog, vao, tf, true);
+wtu.glErrorShouldBe(gl, gl.INVALID_OPERATION, "buffer used as uniform buffer and tf simultaneously");
+gl.bindBufferBase(gl.UNIFORM_BUFFER, 0, uniformBuffer);
+drawWithFeedbackBound(gl, prog, vao, tf, false);
+wtu.glErrorShouldBe(gl, gl.NO_ERROR, "tf buffer not used as uniform buffer anymore");
+drawWithFeedbackBound(gl, prog, vao, tf, true);
+wtu.glErrorShouldBe(gl, gl.NO_ERROR, "tf buffer not used as uniform buffer anymore");
+
+gl.bindTransformFeedback(gl.TRANSFORM_FEEDBACK, tf);
+const tfBuffer2 = createBuffer(gl, Float32Array.BYTES_PER_ELEMENT * 4);
+gl.bindBufferBase(gl.TRANSFORM_FEEDBACK_BUFFER, 0, tfBuffer2);
+drawWithFeedbackBound(gl, prog, badVao, tf);
+wtu.glErrorShouldBe(gl, gl.NO_ERROR, "buffer is no longer bound for transform feedback");
+gl.bindTransformFeedback(gl.TRANSFORM_FEEDBACK, tf);
+gl.bindBufferBase(gl.TRANSFORM_FEEDBACK_BUFFER, 0, tfBuffer);
+
+debug("<hr/>Test PIXEL_UNPACK_BUFFER");
+const tex = gl.createTexture();
+gl.bindTexture(gl.TEXTURE_2D, tex);
+gl.bindBuffer(gl.PIXEL_UNPACK_BUFFER, vertexBuffer);
+gl.texImage2D(gl.TEXTURE_2D, 0, gl.RGBA8, 1, 1, 0, gl.RGBA, gl.UNSIGNED_BYTE, 0);
+wtu.glErrorShouldBe(gl, gl.NO_ERROR, "PIXEL_UNPACK_BUFFER is not bound for transform feedback");
+
+gl.bindBuffer(gl.PIXEL_UNPACK_BUFFER, tfBuffer);
+gl.texImage2D(gl.TEXTURE_2D, 0, gl.RGBA8, 1, 1, 0, gl.RGBA, gl.UNSIGNED_BYTE, 0);
```

Comment:
Maybe add a helper function which takes these operations in an anonymous function, calling them twice, and the second time, calls bindTransformFeedback(gl.TRANSFORM_FEEDBACK, null) first, and assert that the operation succeeds? This will test the code path which does the deref of the buffer's state that it's bound to the active transform feedback object.